src/storage/index.js

```diff
@@ -28,14 +34,22 @@ const StorageOptionsType = {
 /**
  * factory for storage, depending on type
  * @param {StorageOptionsType} type
- * @param {StorageMemoryOptions|StorageRedisOptions} options
+ * @param {StorageMemoryOptions|StorageRedisOptions|StorageCustomOptions} options
  * @returns {StorageMemory|StorageRedis}
  */
 function createStorage (type, options) {
   if (!isServerSide && type === StorageOptionsType.redis) {
     throw new Error('Redis storage is not supported in the browser')
   }
 
+  if (type === 'custom') {
+    if (!options.storage) {
+      throw new Error('Storage is required for custom storage type')
```

Comment:
check the storage instance has the requried methods

Reply:
Hi @simone-sanfratello , The interface class has default error throwing functions. So, if storage doesn't implement a specific function, cache will hit the default error, that's why I didn’t add function-level checks.

Let me know if you want those checks explicitly. Also, I’ve added a test case for an empty storage class.

Reply:
A custom storage implementation could not implement the StorageInterface and having different methods